networkx/algorithms/community/modularity_max.py

```diff
@@ -67,6 +72,13 @@ def greedy_modularity_communities(G, weight=None, resolution=1):
     m = sum([d.get("weight", 1) for u, v, d in G.edges(data=True)])
     q0 = 1.0 / (2.0 * m)
 
+    if n_communities > N:
+        warnings.warn(
+            "Input n_communities is greater than number of nodes of G."
+            "Falling back to default n_communities = 1."
+        )
+        n_communities = 1
+
```

Comment:
```suggestion
    if (n_communities < 1) or (n_communities > N):
        raise ValueError(
            f"n_communities must be between 1 and len(G.nodes()). Got {n_communities}"
        )
```
Two things here:
 1. I think raising an exception is cleaner than the warn-and-set-default approach.
 2. I think we also need a check on the lower bound of n_communities, otherwise it seems we might hit an infinite loop condition

Reply:
Agree! Didn't use direct commit suggestion option, because some brackets were missing, I think: `{len(G.nodes())})`